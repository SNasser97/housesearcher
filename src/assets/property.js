// this file is no longer used as we call it from a server
// this file was just for basic demonstration
export const property = [
	{
    "id":1,
    "house_address":"70 St Denys Road",
    "house_area":"Potter Heighham",
    "house_postcode":"NR29 9FD",
    "house_price":"187,495",
    "house_type":"Terraced",
    "house_desc":"A newly refurbished 4-bedroom family home in Potter Heighham, A lovely terraced property close to the city center for all your shopping needs as well as a great family home and close to many local schools ideal for stress-free school runs",
    "house_features":
      {
        "bedrooms":4,
        "city_distance":0.4,
        "school_distance":0.34,
        "parking":"residential"
      },
    "imageData":"https://s3-eu-west-2.amazonaws.com/homebuilding-assets/prodwebsite/content/uploads/2013/09/Budget-Victorian-Restoration-Gooch.jpg"
  },
  {
    "id":2,
    "house_address":"63 Colorado Way",
    "house_area":"RHUMACH",
    "house_postcode":"PH39 0FZ",
    "house_price":"287,375",
    "house_type":"Terraced",
    "house_desc":"A newly refurbished 5-bedroom family home in Rhumach, great fit for a large family and ideal spot as property is close to a local school and nearby to a secondary school",
    "house_features":
      {
        "bedrooms":5,
        "city_distance":0.8,
        "school_distance":0.24,
        "parking":"garage parking"
      },
    "imageData":"https://odis.homeaway.com/odis/listing/6de14dc6-d119-496c-bc0e-a49d92ec04d8.c10.jpg"
  },
  {
    "id":3,
    "house_address":"123 Park End St",
    "house_area":"BROKEN CROSS",
    "house_postcode":"CW9 5GG",
    "house_price":"134,175",
    "house_type":"Terraced",
    "house_desc":"A well cared property with 3 bedrooms a great home for first-time buyers!",
    "house_features":
      {
        "bedrooms":3,
        "city_distance":2.23,
        "school_distance":0.54,
        "parking":"on-street"
      },
    "imageData":"http://cdn.overstreet.co.uk/images/Property/large/1/2014/12/26/3-bed-terraced-house-for-sale-id14300601.jpg"
  },

  {
    "id":4,
    "house_address":"20 Scotsburn Rd",
    "house_area":"TAMWORTH",
    "house_postcode":"B79 3YA",
    "house_price":"254,975",
    "house_type":"Semi-detached",
    "house_desc":"Lovely semi-detached house with 4 bedrooms a great home for new or upcoming families!",
    "house_features":
      {
        "bedrooms":4,
        "city_distance":1.03,
        "school_distance":0.61,
        "parking":"on-street"
      },
    "imageData":"https://www.propertypriceadvice.co.uk/wp-content/uploads/2019/08/Semi-Detached-Home-UK-RS.jpg"
  },

  {
    "id":5,
    "house_address":"66 Lammas Street",
    "house_area":"GREASBROUGH",
    "house_postcode":"S61 2WB",
    "house_price":"460,000",
    "house_type":"Semi-detached",
    "house_desc":"Brand new home build just completed this year with the top range and latest kitchen and bathrooms installed",
    "house_features":
      {
        "bedrooms":6,
        "city_distance":5.03,
        "school_distance":1.21,
        "parking":"garage parking"
      },
    "imageData":"https://media.rightmove.co.uk/dir/2k/1920/81635810/1920_NML104768_IMG_11_0000_max_656x437.JPG"
  },

   {
    "id":6,
    "house_address":"110 Grenoble Road",
    "house_area":"BRIDEKIRK",
    "house_postcode":"CA13 4NT",
    "house_price":"95,000",
    "house_type":"Semi-detached",
    "house_desc":"New property built in an upcoming area, ideal for first time buyers or settlers. Not too far from city center as only 10min drive away",
    "house_features":
      {
        "bedrooms":2,
        "city_distance":4.28,
        "school_distance":0.86,
        "parking":"on-street"
      },
    "imageData":"https://www.livewest.co.uk/sites/default/files/styles/house_main/public/2019-05/StClareStreetScenePlots72-75-1024x603.jpg?itok=8oXQHimy"
  },


  {
    "id":7,
    "house_address":"91 Thornton St",
    "house_area":"HUNT",
    "house_postcode":"L25 7SP",
    "house_price":"424,758",
    "house_type":"Detached",
    "house_desc":"Perfect home based in a quiet area of the country, ideal for those escaping from the bustling city centers!",
    "house_features":
      {
        "bedrooms":4,
        "city_distance":8.28,
        "school_distance":0.96,
        "parking":"garage parking"
      },
    "imageData":"https://www.ashdownjones.co.uk/wp-content/uploads/2017/12/872.jpeg"
  },
  {
    "id":8,
    "house_address":"84 Witney Way",
    "house_area":"KNAPWELL",
    "house_postcode":"CB3 9SB",
    "house_price":"204,758",
    "house_type":"Detached",
    "house_desc":"Quite residential area with access to local schools and 15-20 minutes away from the city center with decent transport links",
    "house_features":
      {
        "bedrooms":3,
        "city_distance":11.56,
        "school_distance":0.63,
        "parking":"garage parking"
      },
    "imageData":"http://cdn.overstreet.co.uk/images/Property/large/4/2015/3/30/chilcombe-drive-telford-000023558_5512490_IMG_00.jpg"
  },

  
  {
    "id":9,
    "house_address":"57 Horsefair Green",
    "house_area":"ONICH",
    "house_postcode":"PH33 5ZP",
    "house_price":"221,000",
    "house_type":"Detached",
    "house_desc":"Quite residential area with access to local schools and with only a 10 minute drive to city center for all your shopping needs!",
    "house_features":
      {
        "bedrooms":4,
        "city_distance":7.24,
        "school_distance":0.43,
        "parking":"garage parking"
      },
    "imageData":"https://i.dailymail.co.uk/i/pix/2016/06/30/10/35C82C8C00000578-3665582-It_s_rare_to_find_a_large_modern_detached_house_with_bedrooms_fo-a-3_1467278886078.jpg"
  },
  {
    "id":10,
    "house_address":"90 Bishopthorpe Road",
    "house_area":"PENHALE",
    "house_postcode":"TR12 5EB",
    "house_price":"55,963",
    "house_type":"Bungalow",
    "house_desc":"A cosy 2 bedroom bungalow in a calm area ideal for an elderly couple or for those wanting to escape bustling areas. Within walking distance to local town center",
    "house_features":
      {
        "bedrooms":2,
        "city_distance":15.24,
        "school_distance":1.43,
        "parking":"residential"
      },
    "imageData":"https://odis.homeaway.com/odis/listing/84ba7492-ec75-462e-aac1-41516f36f11b.c10.jpg"
  },
  {
    "id":11,
    "house_address":"50 Caxton Place",
    "house_area":"BUSHEY",
    "house_postcode":"WD1 5LH",
    "house_price":"235,000",
    "house_type":"Bungalow",
    "house_desc":"Newly built 3 bedroom bungalow within great walking distance to town center which provides great transport links to city center",
    "house_features":
      {
        "bedrooms":3,
        "city_distance":13.24,
        "school_distance":1.23,
        "parking":"residential"
      },
    "imageData":"https://www.muirhomes.co.uk/wp-content/uploads/2014/10/2016-01-A-Muir-Houses-HT-Brodick-and-Fraser-V3-WC.jpg"
  },
  {
    "id":12,
    "house_address":"49 Haslemere Road",
    "house_area":"EBFORD",
    "house_postcode":"EX3 5ZT",
    "house_price":"202,453",
    "house_type":"Bungalow",
    "house_desc":"Newly built 3 bedroom bungalow within great walking distance to town center which provides great transport links to city center",
    "house_features":
      {
        "bedrooms":3,
        "city_distance":16.25,
        "school_distance":1.45,
        "parking":"residential"
      },
    "imageData":"https://s3.eu-west-2.amazonaws.com/standoutproperty/Media/20ef5f60-732e-4d2d-8a06-f24dc24eda95/8e6c37a0-8805-4ea6-af9b-668f82c50de2/hag-gp210gu4vu-3833fe6fcedf748e1fc2990fe8aef4cf.JPG"
  }
];