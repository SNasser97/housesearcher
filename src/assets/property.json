[
  {
    "id": 1,
    "house_address": "70 St Denys Road",
    "house_area": "Potter Heighham",
    "house_postcode": "NR29 9FD",
    "house_price": "187,495",
    "house_type": "Terraced",
    "house_desc": "A newly refurbished 4-bedroom family home in Potter Heighham, A lovely terraced property close to the city center for all your shopping needs as well as a great family home and close to many local schools ideal for stress-free school runs",
    "house_features": {
      "bedrooms": 4,
      "city_distance": 0.4,
      "school_distance": 0.34,
      "parking": "residential"
    },
    "imageData": "https://i.ibb.co/1mSpYqF/house-1-min.jpg"
  },
  {
    "id": 2,
    "house_address": "63 Colorado Way",
    "house_area": "RHUMACH",
    "house_postcode": "PH39 0FZ",
    "house_price": "287,375",
    "house_type": "Terraced",
    "house_desc": "A newly refurbished 5-bedroom family home in Rhumach, great fit for a large family and ideal spot as property is close to a local school and nearby to a secondary school",
    "house_features": {
      "bedrooms": 5,
      "city_distance": 0.8,
      "school_distance": 0.24,
      "parking": "garage parking"
    },
    "imageData": "https://i.ibb.co/Ch0TT9f/house-2-min.jpg"
  },
  {
    "id": 3,
    "house_address": "123 Park End St",
    "house_area": "BROKEN CROSS",
    "house_postcode": "CW9 5GG",
    "house_price": "134,175",
    "house_type": "Terraced",
    "house_desc": "A well cared property with 3 bedrooms a great home for first-time buyers!",
    "house_features": {
      "bedrooms": 3,
      "city_distance": 2.23,
      "school_distance": 0.54,
      "parking": "on-street"
    },
    "imageData": "https://i.ibb.co/hW4Y2rL/house-3-min.jpg"
  },
  {
    "id": 4,
    "house_address": "20 Scotsburn Rd",
    "house_area": "TAMWORTH",
    "house_postcode": "B79 3YA",
    "house_price": "254,975",
    "house_type": "Semi-detached",
    "house_desc": "Lovely semi-detached house with 4 bedrooms a great home for new or upcoming families!",
    "house_features": {
      "bedrooms": 4,
      "city_distance": 1.03,
      "school_distance": 0.61,
      "parking": "on-street"
    },
    "imageData": "https://i.ibb.co/G25VK34/house-4-min.jpg"
  },
  {
    "id": 5,
    "house_address": "66 Lammas Street",
    "house_area": "GREASBROUGH",
    "house_postcode": "S61 2WB",
    "house_price": "460,000",
    "house_type": "Semi-detached",
    "house_desc": "Brand new home build just completed this year with the top range and latest kitchen and bathrooms installed",
    "house_features": {
      "bedrooms": 6,
      "city_distance": 5.03,
      "school_distance": 1.21,
      "parking": "garage parking"
    },
    "imageData": "https://i.ibb.co/YcSxvNC/house-5-min.jpg"
  },
  {
    "id": 6,
    "house_address": "110 Grenoble Road",
    "house_area": "BRIDEKIRK",
    "house_postcode": "CA13 4NT",
    "house_price": "95,000",
    "house_type": "Semi-detached",
    "house_desc": "New property built in an upcoming area, ideal for first time buyers or settlers. Not too far from city center as only 10min drive away",
    "house_features": {
      "bedrooms": 2,
      "city_distance": 4.28,
      "school_distance": 0.86,
      "parking": "on-street"
    },
    "imageData": "https://i.ibb.co/Jn7Y2wZ/house-6-min.jpg"
  },
  {
    "id": 7,
    "house_address": "91 Thornton St",
    "house_area": "HUNT",
    "house_postcode": "L25 7SP",
    "house_price": "424,758",
    "house_type": "Detached",
    "house_desc": "Perfect home based in a quiet area of the country, ideal for those escaping from the bustling city centers!",
    "house_features": {
      "bedrooms": 4,
      "city_distance": 8.28,
      "school_distance": 0.96,
      "parking": "garage parking"
    },
    "imageData": "https://i.ibb.co/1X7V1FF/house-7-min.jpg"
  },
  {
    "id": 8,
    "house_address": "84 Witney Way",
    "house_area": "KNAPWELL",
    "house_postcode": "CB3 9SB",
    "house_price": "204,758",
    "house_type": "Detached",
    "house_desc": "Quite residential area with access to local schools and 15-20 minutes away from the city center with decent transport links",
    "house_features": {
      "bedrooms": 3,
      "city_distance": 11.56,
      "school_distance": 0.63,
      "parking": "garage parking"
    },
    "imageData": "https://i.ibb.co/r5t6CLc/house-8-min.jpg"
  },
  {
    "id": 9,
    "house_address": "57 Horsefair Green",
    "house_area": "ONICH",
    "house_postcode": "PH33 5ZP",
    "house_price": "221,000",
    "house_type": "Detached",
    "house_desc": "Quite residential area with access to local schools and with only a 10 minute drive to city center for all your shopping needs!",
    "house_features": {
      "bedrooms": 4,
      "city_distance": 7.24,
      "school_distance": 0.43,
      "parking": "garage parking"
    },
    "imageData": "https://i.ibb.co/r3GjFwb/house-9-min.jpg"
  },
  {
    "id": 10,
    "house_address": "90 Bishopthorpe Road",
    "house_area": "PENHALE",
    "house_postcode": "TR12 5EB",
    "house_price": "55,963",
    "house_type": "Bungalow",
    "house_desc": "A cosy 2 bedroom bungalow in a calm area ideal for an elderly couple or for those wanting to escape bustling areas. Within walking distance to local town center",
    "house_features": {
      "bedrooms": 2,
      "city_distance": 15.24,
      "school_distance": 1.43,
      "parking": "residential"
    },
    "imageData": "https://i.ibb.co/HYnY82V/house-10-min.jpg"
  },
  {
    "id": 11,
    "house_address": "50 Caxton Place",
    "house_area": "BUSHEY",
    "house_postcode": "WD1 5LH",
    "house_price": "235,000",
    "house_type": "Bungalow",
    "house_desc": "Newly built 3 bedroom bungalow within great walking distance to town center which provides great transport links to city center",
    "house_features": {
      "bedrooms": 3,
      "city_distance": 13.24,
      "school_distance": 1.23,
      "parking": "residential"
    },
    "imageData": "https://i.ibb.co/mFzMp3V/house-11-min.jpg"
  },
  {
    "id": 12,
    "house_address": "49 Haslemere Road",
    "house_area": "EBFORD",
    "house_postcode": "EX3 5ZT",
    "house_price": "202,453",
    "house_type": "Bungalow",
    "house_desc": "Newly built 3 bedroom bungalow within great walking distance to town center which provides great transport links to city center",
    "house_features": {
      "bedrooms": 3,
      "city_distance": 16.25,
      "school_distance": 1.45,
      "parking": "residential"
    },
    "imageData": "https://i.ibb.co/MDNj1Bf/house-12-min.jpg"
  }
]
