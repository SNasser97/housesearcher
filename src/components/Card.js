import React from 'react';


// props destructure {name,addres...} = props
const Card = ({features,id,name,area, postcode, type, desc, price, image}) => {

	return (
			<div className="card">
				<div className="card__house card__house--front">
					<div className="house__details">
					<img className="house__img" src={image} alt={`property-${id}, ${name}`}/>
						<h2 className="house__name">{name}</h2>
						<p className="house__address">{`${area}, ${postcode}`}</p>
						<p className="house__type">{type}</p>
						<p className="house__price">
							<span className="house__price--currency">Â£</span>
							{price}
						</p>
					</div>
					{/* Flip over and show these details */}
					<div className="card__house card__house--back">
						<div className="house__details">
							<p className="house__description">{desc}
							</p>
							<ul className="house__list">
								<li className="house__item house__item--emojiBed">{`${features["bedrooms"]} bedrooms`}</li>
								<li className="house__item house__item--emojiCity">{`${features["city_distance"]}mi from city center`}</li>
								<li className="house__item house__item--emojiSchool">{`${features["school_distance"]}mi from local school`}</li>
								<li className="house__item house__item--emojiParking">{`parking, ${features["parking"]}`}</li>
							</ul>
							<a className="btn btn__primary" href="#!" target="_blank">View</a>
						</div>

					</div>
				</div>
			</div>
	)
}
export default Card;