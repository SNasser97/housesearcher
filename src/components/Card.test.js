import React from 'react';
import { shallow, mount, render } from 'enzyme';
// import { create } from "react-test-renderer";
import Card from './Card';
import property from '../assets/property.json';

// dummy_data from api (obj data that includes 4 objects)
const {
  house_features,
  house_address,
  house_area,
  house_postcode,
  house_type,
  house_desc,
  house_price,
  imageData,
  id
} = property[0];

const card = shallow(<Card
  features={house_features}
  id={id}
  name={house_address}
  area={house_area}
  postcode={house_postcode}
  type={house_type}
  desc={house_desc}
  price={house_price}
  image={imageData}
/>);
//* snapshot test example
it('expect to render card component', () => {
  expect(card).toMatchSnapshot();
});

// describe('<Card props={obj}...house_features={obj.obj.props} />', () => {
//   const wrapper = shallow(<Card features={features.house_features} />);

//   it('expect to render card', () => {
//     expect(wrapper.length)
//       .toEqual(1);
//   });

//   it('expect to contain description p element', () => {
//     const wrapper = shallow(<Card features = {features.house_features}>
//       <p className='house__description'></p>
//     </Card>);
//     expect(wrapper.contains(<p className='house__description'></p>
// )).toEqual(true);
//     console.log(wrapper.children());
//   });


// });

