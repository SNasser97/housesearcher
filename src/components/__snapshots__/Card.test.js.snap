// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component 1`] = `
<div
  className="card"
>
  <div
    className="card__house card__house--front"
  >
    <div
      className="house__details"
    >
      <img
        alt="property-1, 70 St Denys Road"
        className="house__img"
        height="225"
        src="https://i.ibb.co/1mSpYqF/house-1-min.jpg"
        width="320"
      />
      <h2
        className="house__name"
      >
        70 St Denys Road
      </h2>
      <p
        className="house__address"
      >
        Potter Heighham, NR29 9FD
      </p>
      <p
        className="house__type"
      >
        Terraced
      </p>
      <p
        className="house__price"
      >
        <span
          className="house__price--currency"
        >
          Â£
        </span>
        187,495
      </p>
    </div>
    <div
      className="card__house card__house--back"
    >
      <div
        className="house__details"
      >
        <p
          className="house__description"
        >
          A newly refurbished 4-bedroom family home in Potter Heighham, A lovely terraced property close to the city center for all your shopping needs as well as a great family home and close to many local schools ideal for stress-free school runs
        </p>
        <ul
          className="house__list"
        >
          <li
            className="house__item house__item--emojiBed"
          >
            4 bedrooms
          </li>
          <li
            className="house__item house__item--emojiCity"
          >
            0.4mi from city center
          </li>
          <li
            className="house__item house__item--emojiSchool"
          >
            0.34mi from local school
          </li>
          <li
            className="house__item house__item--emojiParking"
          >
            parking, residential
          </li>
        </ul>
        <a
          className="btn btn__primary"
          href="#!"
          target="_blank"
        >
          View
        </a>
      </div>
    </div>
  </div>
</div>
`;
